klippa/src/colr.rs

```diff
@@ -0,0 +1,1708 @@
+            .base_glyph_list()
+            .transpose()
+            .map_err(|_| SubsetError::SubsetTableError(Colr::TAG))?;
+        let subset_to_v0 = downgrade_to_v0(base_glyph_list.as_ref(), plan);
+
+        serialize_v0(self, plan, s, subset_to_v0)
+            .map_err(|_| SubsetError::SubsetTableError(Colr::TAG))?;
+
+        if subset_to_v0 {
+            return Ok(());
+        }
+
+        // set version to 1, format pos = 0
+        s.copy_assign(0, 1_u16);
+
+        // var_store offset pos = 30
+        if let Some(var_store) = self
+            .item_variation_store()
+            .transpose()
+            .map_err(|_| SubsetError::SubsetTableError(Colr::TAG))?
+        {
+            Offset32::serialize_subset(&var_store, s, plan, &plan.colr_varstore_inner_maps, 30)
+                .map_err(|_| SubsetError::SubsetTableError(Colr::TAG))?;
+        }
+
+        // BaseGlyphList offset pos = 14
+        Offset32::serialize_subset(&base_glyph_list.unwrap(), s, plan, (), 14)
+            .map_err(|_| SubsetError::SubsetTableError(Colr::TAG))?;
+
+        //LayerList offset pos = 18
+        if let Some(layer_list) = self
+            .layer_list()
+            .transpose()
+            .map_err(|_| SubsetError::SubsetTableError(Colr::TAG))?
+        {
+            match Offset32::serialize_subset(&layer_list, s, plan, (), 18) {
+                Ok(()) | Err(SerializeErrorFlags::SERIALIZE_ERROR_EMPTY) => (),
+                Err(_) => {
+                    return Err(SubsetError::SubsetTableError(Colr::TAG));
+                }
+            }
+        }
+
+        //ClipList offset pos = 22
+        if let Some(clip_list) = self
+            .clip_list()
+            .transpose()
+            .map_err(|_| SubsetError::SubsetTableError(Colr::TAG))?
+        {
+            // cliplist could be empty after subsetting
+            match Offset32::serialize_subset(&clip_list, s, plan, (), 22) {
+                Ok(()) | Err(SerializeErrorFlags::SERIALIZE_ERROR_EMPTY) => (),
+                Err(_) => {
+                    return Err(SubsetError::SubsetTableError(Colr::TAG));
+                }
+            }
+        }
+
+        //varIndexMap offset pos = 26
+        if let Some(var_index_map) = self
+            .var_index_map()
+            .transpose()
+            .map_err(|_| SubsetError::SubsetTableError(Colr::TAG))?
+        {
+            if let Some(deltaset_index_map_subset_plan) =
+                create_deltaset_index_map_subset_plan(plan)
+            {
+                Offset32::serialize_subset(
+                    &var_index_map,
+                    s,
+                    plan,
+                    &deltaset_index_map_subset_plan,
+                    26,
+                )
+                .map_err(|_| SubsetError::SubsetTableError(Colr::TAG))?;
+            }
+        }
+        Ok(())
+    }
+}
+
+// serialize header and V0 tables, format is decided by subset_to_v0 flag
+fn serialize_v0(
```

Comment:
if this has a direct parallel in harfbuzz, it would be useful to link to it here.